doc/users/next_whats_new/EngFormatter-offset.rst

```diff
@@ -0,0 +1,12 @@
+ticker.EngFormatter now computes offset by default
+--------------------------------------------------
+
+:class:`matplotlib.ticker.EngFormatter` has gained the ability to show an offset text near the
+axis. With shared logic with :class:`matplotlib.ticker.ScalarFormatter`, it is capable of
+deciding whether the data qualifies having an offset and show it with an
+appropriate SI quantity prefix, and with the supplied ``unit``.
+
+To enable this new behavior, simply pass ``useOffset=True`` when you instantiate
+:class:`matplotlib.ticker.EngFormatter`. Example is available here_.
+
+.. _here: ../../gallery/ticks/engformatter_offset.html
```

Comment:
> if docs don't build, you probably want something like:
> 
> ```
> :doc:`gallery\ticks\engformatter_offset`
> ```
> 
> [matplotlib.org/devdocs/devel/document.html](https://matplotlib.org/devdocs/devel/document.html#refer-to-other-documents-and-sections)

Thanks! I'm not used to `rst`..

Reply:
> no problem, I think I gave you the wrong slash direction - updated now.

Thanks again.

> Also you don't need the link...but using the plot directive would mean no link. See how the other what's news for examples:
> [`main`/doc/users/next_whats_new](https://github.com/matplotlib/matplotlib/tree/main/doc/users/next_whats_new?rgh-link-date=2024-07-04T15%3A34%3A28Z)

Yes I'm familiar with the `.. plot` directive, I think still that the examples are better place to document this, because the source code is also available without having to download it.

Reply:
Aish that link doesn't work according to CI...

https://output.circle-artifacts.com/output/job/8eb8464f-376d-45ab-ad59-ed86f4b29834/artifacts/0/doc/build/html/users/next_whats_new/EngFormatter-offset.html

Reply:
> > , I think still that the examples are better place to document this,
> 
> To be clear, it's not an either or, it's a thank you very much for the example please also add a plot to the release notes.

I added a `..plot` directive pointing to the same example, as the [devel/document.html](https://matplotlib.org/devdocs/devel/document.html#include-figures-and-files) demonstrate.